benchexec/resources.py

```diff
@@ -99,9 +99,83 @@ def get_cpu_cores_per_run(coreLimit, num_of_threads, my_cgroups, coreSet=None):
         logging.debug("Siblings of cores are %s.", siblings_of_core)
     except ValueError as e:
         sys.exit("Could not read CPU information from kernel: {0}".format(e))
-
+    if nohyperthreading:
+        return _get_cpu_cores_per_run_no_hyperthreading(coreLimit, num_of_threads, allCpus, cores_of_package, siblings_of_core)
     return _get_cpu_cores_per_run0(coreLimit, num_of_threads, allCpus, cores_of_package, siblings_of_core)
 
+def _get_cpu_cores_per_run_no_hyperthreading(coreLimit, num_of_threads, allCpus, cores_of_package, siblings_of_core):
```

Comment:
This method duplicates a lot of work and code of the existing method. This can become problematic, for example if we find a bug in one of the methods, we need to remember to check if the bug also exists in the other method.

Is there a way to reuse more code? For example, one could add a parameter to the existing method, although we need to make sure to not overcomplicate the method. Or can we refactor out parts of the method and reuse them?

Another idea would be to just remove the set of all sibling cores from `allCpus` and call the original method, this should also work, right?

Reply:
@PhilippWendler I have refactored the code as per your suggestion and did the preprocessing to remove all sibling cores from `allCpus` and `cores_of_package` and `siblings_of_core`.
After the preprocessing the core allocation algorithm becomes identical for both `hyperthreading` and `no-hyperthreading` mode so the function does not get complicated.
I did the preprocessing the `_get_cpu_cores_per_run0` function only as we can test this function separately without reading from the machine architecture.